lib/ClangImporter/ImportDecl.cpp

```diff
@@ -9022,11 +9028,18 @@ Decl *ClangImporter::Implementation::importDeclAndCacheImpl(
 
   auto Canon = cast<clang::NamedDecl>(UseCanonicalDecl? ClangDecl->getCanonicalDecl(): ClangDecl);
 
-  if (auto Known = importDeclCached(Canon, version, UseCanonicalDecl)) {
-    if (!SuperfluousTypedefsAreTransparent &&
-        SuperfluousTypedefs.count(Canon))
+  Optional<Decl *> optionalKnown =
+      importDeclCached(Canon, version, UseCanonicalDecl);
+  if (optionalKnown.hasValue()) {
+    auto Known = optionalKnown.getValue();
```

Comment:
Actually, it doesn't really matter if `Known` is a nullptr or not. If it is a nullptr, we can just return it. So you can get rid of the condition below.

Reply:
Done.